### PR TITLE
[SC-4848] Improve timeseries forecasting demo

### PR DESCRIPTION
**Pull request by juanmleng**
## Internal Notes for Reviewers
- Create a new quickstart notebook `quickstart_time_series_full_suite.ipynb` to showcase how to document a time series model.
- This notebook implements a Random Forest model as opposed to an OLS statsmodel, as ML methods seem to be more widely used for time series now.
- The new notebook aims to replace the current `tutorial_time_series_forecasting.ipynb`, which uses several test suites and was a bit convoluted, making it difficult for demos with customers. It also included some tests that were very specific to that notebook, difficult to generalize, and used metrics with `matplotlib` instead of `plotly`.
- All the metrics in the new notebook are implemented with `plotly`, so some have been updated accordingly.
- New metrics have been included to support multiple models.
- Uses the `run_documentation_tests` against this new template [Time Series Forecasting Random Forest](https://app.prod.validmind.ai/settings/templates/clxahgmst01qo29if6wjxkrr2/model_documentation/latest).

## External Release Notes

- **New Quickstart Notebook**: Introduced `quickstart_time_series_full_suite.ipynb` to showcase how to document a time series model.
- **Simplified Example**: Designed to replace the previous `tutorial_time_series_forecasting.ipynb` which has now been removed. The new notebook simplifies the workflow, making it more suitable for demos by removing convoluted data processing, model development steps and the use of not clearly visible test suites.
- **Updated Model**: Implemented both Random Forests and Gradient Boosting models instead of one OLS `statsmodel` model for two different datasets. This change aligns the use case with the increasing use of ML for time series modelling.
- **Enhanced Visualization**: Switched all metrics to use `plotly` instead of `matplotlib`, allowing users to inspect all plots interactively.
- **Support for Multiple Models**: Added new metrics to support the evaluation of multiple models, probabilistic forecasts using confidence intervals and the inspection of model performance across different time segments. 
- **Execution**: All tests are automatically executed using `run_documentation_tests` against the new template [Time Series Forecasting with ML](https://app.prod.validmind.ai/settings/templates/clxahgmst01qo29if6wjxkrr2/model_documentation/latest).

### CODE REVIEW

File: validmind/tests/model_validation/ModelMetadataComparison.py [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+# See the LICENSE file in the root of this repository for details.
+# SPDX-License-Identifier: AGPL-3.0 AND ValidMind Commercial
+
+import pandas as pd
+from validmind.utils import get_model_info
+
+
+def ModelMetadataComparison(models):
+    """
+    Model Metadata Comparison
```
Comment (johnwalz97) [inline validmind/tests/model_validation/ModelMetadataComparison.py:11]: Just a quick heads up that this one doesn't have a full description

File: validmind/tests/model_validation/TimeSeriesPredictionWithCI.py [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+
+import pandas as pd
+import numpy as np
+import plotly.graph_objects as go
+from scipy.stats import norm
+
+
+def TimeSeriesPredictionWithCI(dataset, model, confidence=0.95):
+    """
+    Plot actual vs predicted values for a time series with confidence intervals and compute breaches.
```
Comment (johnwalz97) [inline validmind/tests/model_validation/TimeSeriesPredictionWithCI.py:13]: Same thing here

File: validmind/tests/model_validation/TimeSeriesPredictionsPlot.py [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+# See the LICENSE file in the root of this repository for details.
+# SPDX-License-Identifier: AGPL-3.0 AND ValidMind Commercial
+
+import plotly.graph_objects as go
+import plotly.express as px
+
+
+def TimeSeriesPredictionsPlot(datasets, models):
+    """
+    Plot actual vs predicted values for time series data
```
Comment (johnwalz97) [inline validmind/tests/model_validation/TimeSeriesPredictionsPlot.py:11]: same here :)

File: validmind/tests/model_validation/sklearn/FeatureImportanceComparison.py [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+# See the LICENSE file in the root of this repository for details.
+# SPDX-License-Identifier: AGPL-3.0 AND ValidMind Commercial
+
+import pandas as pd
+from sklearn.inspection import permutation_importance
+
+
+def FeatureImportanceComparison(datasets, models, num_features=3):
+    """
+    Feature Importance Comparison
```
Comment (johnwalz97) [inline validmind/tests/model_validation/sklearn/FeatureImportanceComparison.py:11]: anotha' one

File: validmind/tests/model_validation/sklearn/RegressionErrorsComparison.py [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+from sklearn import metrics
+
+from validmind.logging import get_logger
+
+logger = get_logger(__name__)
+
+
+def RegressionErrorsComparison(datasets, models):
+    """
+    Regression Errors Comparison
```
Comment (johnwalz97) [inline validmind/tests/model_validation/sklearn/RegressionErrorsComparison.py:16]: ➕

File: validmind/tests/model_validation/sklearn/RegressionR2SquareComparison.py [hunk 6]
```diff
@@ -0,0 +1,41 @@
...
+
+import pandas as pd
+from sklearn import metrics
+
+from validmind.tests.model_validation.statsmodels.statsutils import adj_r2_score
+
+
+def RegressionR2SquareComparison(datasets, models):
+    """
+    Regression R2 Square Comparison
```
Comment (johnwalz97) [inline validmind/tests/model_validation/sklearn/RegressionR2SquareComparison.py:13]: ➕

File: notebooks/code_samples/time_series/quickstart_time_series_full_suite.ipynb
Comment (johnwalz97) [inline notebooks/code_samples/time_series/quickstart_time_series_full_suite.ipynb:1]: very nice 👌

**Review by johnwalz97 — APPROVED** [review]
Left a couple comments on some tests that are missing full descriptions but should be good to go other than that 👍

File: validmind/tests/model_validation/ModelMetadataComparison.py (hunk 1 above)
Comment (juanmleng) [reply validmind/tests/model_validation/ModelMetadataComparison.py]: Good call out :)

**Comment by juanmleng** [comment]
I modified the use case slightly to make it more intuitive. The updated notebook uses the same features and two different models: random forest and gradient boosting regressors.

File: notebooks/code_samples/time_series/tutorial_time_series_forecasting.ipynb
Comment (cachafla) [inline notebooks/code_samples/time_series/tutorial_time_series_forecasting.ipynb:1]: Can this notebook be deleted? It still uses time series.

**Review by cachafla — APPROVED** [review]
Awesome 💪 I left a couple of comments.
